packages/building-rollup/src/utils.js

```diff
@@ -39,11 +40,30 @@ function pluginWithOptions(plugin, userConfig, defaultConfig, ...otherParams) {
   return plugin(config, ...otherParams);
 }
 
+/**
+ *
+ * @param {SpaOptions} userOptions
+ * @param {string} outputDir
+ * @param {string} htmlFileName
+ */
+function createSwPath(userOptions, outputDir, htmlFileName = '') {
+  let swPath;
+  if (typeof userOptions.workbox === 'object' && userOptions.workbox.swDest) {
+    swPath = userOptions.workbox.swDest.replace(`${outputDir}/`, '');
+  } else {
+    swPath = './sw.js';
+  }
+
+  swPath = path.relative(path.dirname(htmlFileName), swPath);
```

Comment:
I don't think this will work for windows. We need to check which variables are always POSIX, or if some use native file system separators.